### DISCUSSION TITLE
Re-authentication was successful

### DISCUSSION THREAD

**Discussion by davsoliman** [astrandb/miele#557] [Q&A]
Hey guys,

I've managed to follow all instructions however when i go to link the integration to HA it says "Re-authentication was successful" and nothing helps happens from here.

Any advice?

**Comment by igor-sweet** [comment]
Same here

**Comment by astrandb** [comment]
There are som cross-posting  - please try to avoid that. Please see the response below:

"Reauthentication was succesful" indicates that there is an entry already in HA. So something is wrong in the internals.

If your device belongs to semi-professional or professional series there will probably be problems due to incomplete and unreliable support in the Miele API. Make sure that your device is consumer grade. Beta version 2024.9.0b1 of the integration will log a message if an affected model is discovered in your account.

There is probably something corrupt in the installation of the integration in your instance. It is not trivial to find the root cause so I suggest that you remove all traces of the integration and start over.
1. Restart Home Assistant
2. Goto HACS and remove the Miele integration. - Open the three-dot-menu at the Miele entry and select Remove.
3. Restart Home Assistant 
4. Remove application credentials. Goto Settings->Devices&services, open the three dot menu in upper right corner and goto Application credentials.
5. Goto HACS - Enter Miele in the search bar, select Miele integration and click download.
6. Restart Home Assistant
7. If there is a reasonably modern Miele device on your LAN it should be discovered by HA within a few minutes if your LAN allows broadcasts. If the device is not auto-discovered you can add the integration manually by following instructions below.
8. Goto Settings->Devices&services, Click Add Integration and select Miele
9. Follow the instructions to setup the integration.
10. You can see the expected flow here https://github.com/astrandb/miele/discussions/366 . Please report  which step that fails.

**Reply by davsoliman** [reply]
Thanks for your assistance.

I have 3 Miele products that are roughly 4-5 years old, I can see them in the Miele App:

- Dishwasher G6927 SCI XXL
- Washing Machine WRC890 WPS
- Dryer TCR 870 WP

I've removed/installed 3-4 times now following the above steps and I'm still having the same issue.

Once I get to the login page and enter my credentials it takes me to the "link account to home assistant page", I then hit "Link account" and it reverts me back to the home assistant page saying "Re-authentication was successful". 

It doesn't give any options to add any of the above 3 machines I have.

Looking forward to your guidance!

**Reply by astrandb** [reply]
I forgot to mention above that you should also remove the stored application credentials for the Miele intergration. See newly inserted item 4 above.

**Reply by davsoliman** [reply]
Interestingly enough I cant seem to remove my credentials.

After removing Miele > restarted > remove credentials lead to this notification "Application Credentials in use by an integration may not be removed."

Is there something running in the background stopping this?

**Reply by astrandb** [reply]
> Is there something running in the background stopping this?

There is something odd going on. What type of installation of HA are you using? Would it be possible for me to get access to your HA? You can contact me directly at ake@strandberg.eu
